### ISSUE TITLE
Unable to Pull deleted Apex Classes

### ISSUE THREAD

**Issue by vishnuyadavalli** [forcedotcom/devops-center-feedback#204]
**Describe the bug**
We deleted a bunch of Apex classes and Visualforce Pages. But only a few of them are showing up when we pull changes in work item. 

**To Reproduce**
1. Deleted existing Apex classes
2. Click Pull Changes button in Work Item

**Expected behavior**
Only a few classes are showing up.

**Screenshots**
![image](https://user-images.githubusercontent.com/87042823/183966777-37e1f8d2-c84b-4267-a7af-c3537afcdd3c.png)

![image](https://user-images.githubusercontent.com/87042823/183967139-a917e7c9-d4f2-456a-84f3-47abca58ce5c.png)

**Additional context**

**Comment by EstebanRomero84** [comment]
Hello @vishnuyadavalli thanks for reporting this issue, I will start investigating it. Can you send me the orgId of the environment where you have DevOps Center package installed? And just to rule it out it, have you checked the ignored files?

**Comment by vishnuyadavalli** [comment]
Hi @EstebanRomero84 ,

Yes I checked Ignored files. They are not in ignored files. As you can see, only 1 apex class is pulled. But the rest of the classes are not pulled.

Org Id - 00DE0000000YgOr

**Comment by EstebanRomero84** [comment]
Could you run the following query in the dev environment where you deleted the components?
`SELECT Id, MemberName, MemberType, IsNameObsolete, IsNewMember, RevisionCounter, LastModifiedDate FROM SourceMember ORDER BY RevisionCounter DESC`
You will need to select **Use Tooling API**

**Comment by vishnuyadavalli** [comment]
@EstebanRomero84 ,

Can find the deleted apex classes with the query.
![image](https://user-images.githubusercontent.com/87042823/184005152-69d7a642-09b3-43fa-9ca1-dd4a132577ce.png)

**Comment by EstebanRomero84** [comment]
Thank you @vishnuyadavalli this is very helpful.
Another question, how did you delete those classes? From setup, from dev console, sfdx, destructive deployment, or other way?

**Comment by vishnuyadavalli** [comment]
@EstebanRomero84 some from Dev Console, some form Setup ->Apex classes.

**Comment by EstebanRomero84** [comment]
@vishnuyadavalli the org where you deleted the classes is a sandbox or a scratch org? And if the answer is sandbox, which type and when source tracking was enabled on it?

Thanks

**Comment by vishnuyadavalli** [comment]
@EstebanRomero84 Yes. It is a sandbox and created after source tracking is enabled.

**Comment by EstebanRomero84** [comment]
@vishnuyadavalli the issue here seems to be related to the Source Member object, which DevOps Center uses to track changes. For some reason Source Member records were not created when you deleted those classes. We contacted internally the team that owns this functionality and they aware of this issue. You can track it [here](https://github.com/forcedotcom/cli/issues/1622).

**Comment by vishnuyadavalli** [comment]
@EstebanRomero84 . Thanks you.

**Comment by amtrack** [comment]
Is there an update or Known Issue for it?

I rarely work with Sandboxes and when this was reported to me I was surprised.

From what I can see, this is how Source Tracking works in Sandboxes:

- Refresh or create a Sandbox
- The `SourceMember` Tooling API object does not have any records (check with: `sf data query -t -q "SELECT MemberType, MemberName, IsNameObsolete FROM SourceMember"`)
- Create new Metadata or update existing Metadata -> a `SourceMember` record get's created automatically  ✅
- Delete existing Metadata: -> no `SourceMember` record get's created automatically ❌
